oops/backplane/exercise_backplanes.py

```diff
@@ -144,872 +34,224 @@ def show_info(title, array):
-    show_info('Mask of Saturn sunward', test)
-
-    test = bp.evaluate(('where_sunward', 'saturn'))
-    show_info('Mask of Saturn sunward via evaluate()', test)
-
-    test = bp.where_below(('incidence_angle', 'saturn'), HALFPI)
-    show_info('Mask of Saturn sunward via where_below()', test)
-
-    test = bp.where_antisunward('saturn')
-    show_info('Mask of Saturn anti-sunward', test)
-
-    test = bp.where_above(('incidence_angle', 'saturn'), HALFPI)
-    show_info('Mask of Saturn anti-sunward via where_above()', test)
-
-    test = bp.where_between(('incidence_angle', 'saturn'), HALFPI,3.2)
-    show_info('Mask of Saturn anti-sunward via where_between()', test)
-
-    test = bp.where_in_front('saturn', 'saturn_main_rings')
-    show_info('Mask of Saturn in front of rings', test)
-
-    test = bp.where_in_back('saturn', 'saturn_main_rings')
-    show_info('Mask of Saturn behind rings', test)
-
-    test = bp.where_inside_shadow('saturn', 'saturn_main_rings')
-    show_info('Mask of Saturn in shadow of rings', test)
-
-    test = bp.where_outside_shadow('saturn', 'saturn_main_rings')
-    show_info('Mask of Saturn outside shadow of rings', test)
-
-    test = bp.where_intercepted('saturn_main_rings')
-    show_info('Mask of rings intercepted', test)
-
-    test = bp.where_in_front('saturn_main_rings', 'saturn')
-    show_info('Mask of rings in front of Saturn', test)
-
-    test = bp.where_in_back('saturn_main_rings', 'saturn')
-    show_info('Mask of rings behind Saturn', test)
-
-    test = bp.where_inside_shadow('saturn_main_rings', 'saturn')
-    show_info('Mask of rings in shadow of Saturn', test)
-
-    test = bp.where_outside_shadow('saturn_main_rings', 'saturn')
-    show_info('Mask of rings outside shadow of Saturn', test)
-
-    test = bp.where_sunward('saturn_main_rings')
-    show_info('Mask of rings sunward', test)
-
-    test = bp.where_antisunward('saturn_main_rings')
-    show_info('Mask of rings anti-sunward', test)
-
-    ########################
-
-    if printing: print('\n********* borders')
+    config.LOGGING.off()
 
-    mask = bp.where_intercepted('saturn')
-    test = bp.border_inside(mask)
-    show_info('Border of Saturn intercepted mask, inside', test)
 
-    test = bp.border_outside(mask)
-    show_info('Border of Saturn intercepted mask, outside', test)
 
-    test = bp.border_below(('ring_radius', 'saturn:ring'), 100.e3)
-    show_info('Border of ring radius below 100 kkm', test)
+#===============================================================================
+def exercise_backplanes_settings(obs):
+    """Wrapper for exercise_backplanes(). Determines default directories
+       based on input observation object and fills in args based on
+       settings."""
 
-    test = bp.border_atop(('ring_radius', 'saturn:ring'), 100.e3)
-    show_info('Border of ring radius atop 100 kkm', test)
+    # determine default output and reference directories
+    specname = obs.filespec.split(TESTDATA_PARENT_DIRECTORY)[1]
```

Comment:
This doesn't work for me and causes all of the backplane tests to fail. `obs.filespec` is something like `/test_dir/cassini/ISS/file.IMG` then `specname` is `/cassini/ISS/file.IMG` then `basedir` is `/cassini/ISS`. Then the creation of `outdir` thinks that `basedir` is an absolute path so it ignores all of the items in the path before than. Is it possible you're making your TESTDATA_PARENT_DIRECTORY with a slash at the end? This should not be necessary and the code should handle both cases.

Reply:
Yes, I have my $OOPS_TEST_DATA_PATH defined with a trailing /, so this make sense.  I'll use os.path.dirname() to standardize TESTDATA_PARENT_DIRECTORY before splitting the path.

Reply:
this should be fixed now, so go ahead and try it and let me know